src/wms/src/main/java/org/geoserver/wms/map/PDFMapResponse.java

```diff
@@ -225,4 +258,159 @@ public void write(Object value, OutputStream output, Operation operation) throws
             throw new ServiceException("Error setting up the PDF", t, "internalError");
         }
     }
+    
+    private static class PDFStreamingRenderer extends StreamingRenderer {
+        
+        public PDFStreamingRenderer() {
+            this.painter = new PDFStyledPainter(labelCache);
```

Comment:
Not available, not should not be made available. The painter is not supposed to be replaced, as you can see from the code this is really an advanced change (e.g., you exactly know what you're doing, or boom). So best if it's hard and not very visible, I don't want to give users ideas about it.